store.go

```diff
@@ -3688,17 +3688,28 @@ func copyStringSlice(slice []string) []string {
 	if len(slice) == 0 {
 		return nil
 	}
-	ret := make([]string, len(slice))
-	copy(ret, slice)
-	return ret
+	return slices.Clone(slice)
+}
+
+func copyStringInt64Map(m map[string]int64) map[string]int64 {
+	if m == nil {
+		return make(map[string]int64, len(m))
+	}
+	return maps.Clone(m)
+}
+
+func copyStringDigestMap(m map[string]digest.Digest) map[string]digest.Digest {
+	if m == nil {
+		return make(map[string]digest.Digest, len(m))
+	}
+	return maps.Clone(m)
 }
```

Comment:
If these functions need to be added at all:
- Please use a generic function instead of duplicating it
- Name and document the `nil` behavior clearly, so that callers see why `maps.Clone` is not being used. `createNonNilMapCloning` is a horrible name, there must be a better option – but something expressing that kind of thing at the call site.